### PR TITLE
TMDM-14625 Foreign Key Item not set to value entered in the Web UI

### PR DESCRIPTION
**Pull request by htyin**
**What is the current behavior?** (You should also link to an open issue here)
FK doesn't update correctly when refer to itself or 2 FK fields refer to the same entity

**What is the new behavior?**
Getting the correct old value for FK field when the field refer to itself or the same entity with another field.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by htyin** [comment]
http://192.168.31.210:8080/view/Master/job/Master_03_TMDMEE_FOR-JIRA/416/

File: org.talend.mdm.core/src/com/amalto/core/history/accessor/record/SimpleValue.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public void set(MetadataRepository repository, DataRecord record, PathElement el
             ReferenceFieldMetadata fieldMetadata = (ReferenceFieldMetadata) element.field;
             boolean needResetValue = true;
             if (record.get(element.field) != null) {
-                String oldValue = String.valueOf(record.get(fieldMetadata.getReferencedField()));
+                String oldValue = String.valueOf(DataRecord.getId((DataRecord) record.get(element.field)));
```
Comment (hw-zhu) [inline org.talend.mdm.core/src/com/amalto/core/history/accessor/record/SimpleValue.java:34]: I'm not sure if DataRecord.getId() can be used for all of field? including FK, complex type ref, composite keys or empty value if no key is defined.

Comment (htyin) [reply org.talend.mdm.core/src/com/amalto/core/history/accessor/record/SimpleValue.java:34]: We have filter above, ReferenceFieldMetadata and not null

File: org.talend.mdm.core.storage.sql/src/test/java/com/amalto/core/save/DocumentSaveTest.java [hunk 2]
```diff
@@ -4743,6 +4743,39 @@ public void test13587_SaveForeignKeyForReusableType() throws Exception {
...
         assertEquals("[11]", evaluate(committedElement, "/TestC/DocterField/BaseField/TestA_FK"));
     }
+    
+    public void test_UpdateSpecialFK() throws Exception {
+    	MetadataRepository repository = new MetadataRepository();
+        repository.load(DocumentSaveTest.class.getResourceAsStream("testSpecialFK.xsd"));
+        MockMetadataRepositoryAdmin.INSTANCE.register("testSpecialFK", repository);
+        SaverSource source = new TestSaverSource(repository, true, "testSpecialFK_original.xml", "testSpecialFK.xsd");
+
+        //Case 1 update foreign key refer to it self
```
Comment (lpw82gf) [inline org.talend.mdm.core.storage.sql/src/test/java/com/amalto/core/save/DocumentSaveTest.java:4753]: it self -> itself
format the comment line

**Comment by htyin** [comment]
http://192.168.31.210:8080/view/Master/job/MDM-REST-EE-MySQL8-CentOS-DEBUG2/649/
http://192.168.31.210:8080/view/Master/job/MDM-SOAP-EE-CentOS-MySQL8-DEBUG1/551/#showFailuresLink

**Pull request merged**